coral-service/src/main/java/com/linkedin/coral/coralservice/metastore/MetastoreProvider.java

```diff
@@ -0,0 +1,125 @@
+package com.linkedin.coral.coralservice.metastore;
+
+import com.linkedin.coral.common.HiveMetastoreClient;
+import com.linkedin.coral.common.HiveMscAdapter;
+import com.linkedin.coral.coralservice.utils.CoralUtils;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+import javax.naming.ConfigurationException;
+
+import com.google.common.base.Strings;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.hadoop.hive.metastore.IMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.ql.metadata.Hive;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.ql.session.SessionState;
+import org.apache.hadoop.security.UserGroupInformation;
+
+import static org.apache.hadoop.hive.conf.HiveConf.ConfVars.*;
+
+
+/**
+ * Utility class to get Hive metastore client for local hive or grid based hive using kerberos
+ */
+@SuppressWarnings("WeakerAccess")
+public class MetastoreProvider {
+
+  public static final String HIVE_METASTORE_URIS = "hive.metastore.uris";
+  public static final String HIVE_METASTORE_AUTHENTICATION = "hive.metastore.authentication";
+  public static final String HIVE_METASTORE_CLIENT_PRINCIPAL = "hive.metastore.client.principal";
+  public static final String HIVE_METASTORE_CLIENT_KEYTAB = "hive.metastore.client.keytab";
+  public static final String HIVE_METASTORE_SERVICE_PRINCIPAL = "hive.metastore.service.principal";
+  public static final String HADOOP_SECURITY_AUTHENTICATION = "hadoop.security.authentication";
+
+  // following properties are to create local hive metastore
+  public static final String HIVE_DIR = "hive.dir";
+
+  public static final String DEFAULT_METASTORE_URI = "thrift://localhost:7552";
+  public static final String DEFAULT_METASTORE_AUTHENTICATION = "SIMPLE";
+  public static final String KERBEROS_AUTHENTICATION = "kerberos";
+  public static final String DEFAULT_TMP_HIVE_DIR;
+
+  static {
+    DEFAULT_TMP_HIVE_DIR = Paths.get(System.getProperty("java.io.tmpdir"), "coral").toString();
+  }
+
+  public static HiveMetastoreClient getMetastoreClient() throws Exception {
+    final InputStream hiveConfStream =
+        CoralUtils.class.getClassLoader().getResourceAsStream("hive.properties");
+    final Properties props = new Properties();
+    props.load(hiveConfStream);
+    return new HiveMscAdapter(MetastoreProvider.getGridMetastoreClient(props));
```

Comment:
At present, there is no `hive.properties` file, so it will fail with NPE with clean checkout.
I would recommend:
1. Add a dummy `hive.properties` file
2. If `hive.properties` doesn't exist or fails to instantiate grid HMS client (may use try catch), we can warn the user and fall back to the local HMS.

Reply:
Adding a dummy `hive.properties`. This seems a lot more user friendly.